### PR TITLE
Check if the installed version of Locksmith is recent or out of date

### PR DESCRIPTION
**Pull request by SamErde**
Compares the installed version of Locksmith to the version on GitHub. If older than [x] days (default is 90), it recommends the user to update. If unable to reach GitHub, it extrapolates the installed version's release date and recommends an update if necessary.

Will submit PR to include this check in main function after testing and validation by the team.

### CODE REVIEW

**Comment by SamErde** [comment]
After mostly completing this, I thought it might be better to target the PowerShell Gallery instead of GitHub for version checks. I've worked at orgs that blocked GitHub downloads but didn't actively block the PowerShell Gallery. Thoughts?

**Comment by TrimarcJake** [comment]
Why not both?

**Comment by SamErde** [comment]
Want to minimize the time it takes to run this check on the client.

We'll need to figure out whether to run this automatically or provide a switch to let the user run the check during execution.

**Comment by SamErde** [comment]
... But at this point, it shouldn't be too hard to use the Gallery or GitHub as backup checks for each other in case one check fails.

File: Private/Get-ModuleAge.ps1 [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    param (
+        # Check a specific version number from the script
+        [Parameter()]
+            [string]$Version,
+        # Define the number of days past a module release date at which to consider the release "out of date."
+        [Parameter()]
+            [int16]$Days = 90
+    )
+
+    if ( -not($Version) ) {
```
[the remark this answers is not in the document]
Comment (SamErde) [reply Private/Get-ModuleAge.ps1]: Makes sense. I think when I wrote that I was picturing the op checking themselves, but...why?! Haha. Oh well, it was fun to write.

File: Private/Get-ModuleAge.ps1 [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            $InstalledVersionMonth = [datetime]::Parse($Version.Replace('.','-')+"-01")
+            # Release date is typically the first Saturday of the month
+            $InstalledVersionReleaseDate = $InstalledVersionMonth.AddDays( 6 - ($InstallVersionMonth.DayOfWeek) )
+        } catch {
+            Write-Error -Message "Unable to find an installed version of Locksmith. That's awkward."
+            break
+        }
+    }
+
+    try {
```
[the remark this answers is not in the document]
Comment (SamErde) [reply Private/Get-ModuleAge.ps1]: I like this, and it gives me another idea: if we add a tag to indicate versions that bring new coverage, we could reference that tag in version checks to say "a newer version includes more checks."

**Review by TrimarcJake — APPROVED** [review]
This is awesome. I am going to merge it with testing and then work on moving a few things around in a future release. I also feel like this could be at least 2 but maybe even three functions!

**Pull request merged**